include/envoy/upstream/load_balancer_type.h

```diff
@@ -25,6 +25,14 @@ enum class LoadBalancerType {
   ClusterProvided
 };
 
+struct SubsetSelector {
+  std::set<std::string> selector_keys;
```

Comment:
Is there a reason this set is ordered?

Reply:
@htuch yes, that's how we ensure fast lookups in subset selectors for given metadata match criteria (which is sorted as well), see also https://github.com/envoyproxy/envoy/blob/master/include/envoy/router/router.h#L514